static/app/components/events/interfaces/spans/embeddedSpanTree.tsx

```diff
@@ -46,20 +48,28 @@ export function EmbeddedSpanTree(props: Props) {
         <QuickTraceQuery event={event} location={location} orgSlug={organization.slug}>
           {results => {
             if (results.isLoading) {
-              return null;
+              return <LoadingIndicator />;
+            }
+
+            if (!results.currentEvent) {
+              return <LoadingError />;
             }
 
             return (
               <GenericDiscoverQuery
                 eventView={eventView}
                 orgSlug={organization.slug}
-                route={`events/${projectSlug}:${results.currentEvent?.event_id}/`}
+                route={`events/${projectSlug}:${results.currentEvent?.event_id}`}
                 api={api}
                 location={location}
               >
                 {_results => {
                   if (_results.isLoading) {
-                    return null;
+                    return <LoadingIndicator />;
+                  }
+
+                  if (!_results.tableData) {
+                    return <LoadingError />;
```

Comment:
nit: is there a way to customize the error message? we can say that we couldn't find the span tree data or something like that

Reply:
not a blocker though

Reply:
Yeah we can, but I wasn't creative enough to come up with a quirky message :)
We can add one in later